src/timetable_gui.cpp

```diff
@@ -47,12 +49,58 @@ struct TimetableArrivalDeparture {
  */
 void SetTimetableParams(int param1, int param2, TimerGameTick::Ticks ticks)
 {
-	if (_settings_client.gui.timetable_in_ticks) {
-		SetDParam(param1, STR_TIMETABLE_TICKS);
-		SetDParam(param2, ticks);
-	} else {
-		SetDParam(param1, STR_TIMETABLE_DAYS);
-		SetDParam(param2, ticks / Ticks::DAY_TICKS);
+	switch (_settings_client.gui.timetable_mode) {
+		case TimetableMode::Days:
+			SetDParam(param1, STR_UNITS_DAYS);
+			SetDParam(param2, ticks / Ticks::DAY_TICKS);
+			break;
+		case TimetableMode::Seconds:
+			SetDParam(param1, STR_UNITS_SECONDS);
+			SetDParam(param2, ticks / Ticks::TICKS_PER_SECOND);
+			break;
+		case TimetableMode::Ticks:
+			SetDParam(param1, STR_UNITS_TICKS);
+			SetDParam(param2, ticks);
+			break;
+		default:
+			NOT_REACHED();
+	}
+}
+
+/**
+ * Get the number of ticks in the current timetable display unit.
+ * @return The number of ticks per day, second, or tick, to match the timetable display.
+ */
+static inline TimerGameTick::Ticks TicksPerTimetableUnit()
+{
+	switch (_settings_client.gui.timetable_mode) {
+		case TimetableMode::Days:
+			return Ticks::DAY_TICKS;
+		case TimetableMode::Seconds:
+			return Ticks::TICKS_PER_SECOND;
+		case TimetableMode::Ticks:
+			return 1;
+		default:
+			NOT_REACHED();
+	}
+}
+
+/**
+ * Determine if a vehicle should be shown as late, depending on the timetable display setting.
+ * @param v The vehicle in question.
+ * @return True if the vehicle is later than the threshold.
+ */
+bool VehicleIsAboveLatenessThreshold(const Vehicle *v)
+{
+	switch (_settings_client.gui.timetable_mode) {
+		case TimetableMode::Days:
+			return v->lateness_counter > Ticks::DAY_TICKS;
+		case TimetableMode::Seconds:
+			return v->lateness_counter > Ticks::TICKS_PER_SECOND;
+		case TimetableMode::Ticks:
+			return v->lateness_counter == 0;
```

Comment:
Is this correct? Previously it used the ` > Ticks::DAY_TICKS` condition, now it shows as late if it's ... not late?